### PR TITLE
fix: use networking.k8s.io/v1beta1 apiVersion for ingress if supported

### PR DESCRIPTION
**Pull request by chgl**
Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/argocd-server/ingress.yaml [hunk 1]
```diff
@@ -2,7 +2,11 @@
 {{- $serviceName := include "argo-cd.server.fullname" . -}}
 {{- $servicePort := .Values.server.service.servicePortHttp -}}
 {{- $paths := .Values.server.ingress.paths -}}
+{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
 apiVersion: extensions/v1beta1
+{{- end }}
```
Comment (carlosjgp) [inline charts/argo-cd/templates/argocd-server/ingress.yaml:9]: A more explicit way of evaluating this is using `.Capabilities.APIVersions.Has`

```
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" }}
apiVersion: networking.k8s.io/v1beta1
{{ else }}
apiVersion: extensions/v1beta1
{{ end -}}
```

Comment (chgl) [reply charts/argo-cd/templates/argocd-server/ingress.yaml]: Awesome, didn't know about this. That's definitely clearer.

**Comment by carlosjgp** [comment]
Looks like there are conflicts on your branch.

Other than that LGTM :rocket:

**Comment by chgl** [comment]
@carlosjgp all done. Thank you for your help!

**Comment by carlosjgp** [comment]
> @carlosjgp all done. Thank you for your help!

Waiting for one of the "3 pending reviewers" :point_down:  to approve and merge ;)

**Pull request merged**